packages/rollup/src/plugins/with-nx/with-nx.ts

```diff
@@ -250,14 +250,14 @@ export function withNx(
       image(),
       json(),
       // TypeScript compilation and declaration generation
-      // TODO(v22): Change default value of useLegacyTypescriptPlugin to false for Nx 22
-      options.useLegacyTypescriptPlugin !== false
+      options.useLegacyTypescriptPlugin === true
```

Comment:
Can switching the plugin cause issues or unexpected results? Just to confirm whether we need a migration to keep the previous behavior.